### PR TITLE
Eng 10549 kafka system tests short row counts issue

### PR DESCRIPTION
**Pull request by JadenW**
This should solve following two issues:
1. ImportManager might gets new catalog with new procedure before invocation dispatcher.
    callProcedure will fail with losing kafka records

2. When restarting kafkaStreamimport, channel add assignment may be ignored if one node is still in stopping status.

### CODE REVIEW

File: src/frontend/org/voltdb/importer/ChannelChangeCallback.java [hunk 1]
```diff
@@ -19,6 +19,9 @@
 
 
 public interface ChannelChangeCallback {
+    default public boolean canAcceptAssignments() {
```
Comment (akhanzode) [inline src/frontend/org/voltdb/importer/ChannelChangeCallback.java]: what is default keyword here?

File: src/frontend/org/voltdb/importer/ChannelDistributer.java [hunk 2]
```diff
@@ -430,13 +430,17 @@ public void registerCallback(String importer, ChannelChangeCallback callback) {
...
+                    if (dispatch.canAcceptAssignments()) {
+                        m_buses.submit(new DistributerRunnable() {
+                            @Override
+                            public void susceptibleRun() throws Exception {
+                                dispatch.onChange(assignment);
+                            }
+                        });
+                        itr.remove();
+                    } else {
+                        LOG.debug("importer " + dispatch + " is stopping, not sending the assigment " + assignment);
```
Comment (akhanzode) [inline src/frontend/org/voltdb/importer/ChannelDistributer.java]: should message say "channel assignments will be sent when importer is started"

File: src/frontend/org/voltdb/importer/ImporterLifeCycleManager.java [hunk 3]
```diff
@@ -141,6 +141,12 @@ private void startImporters(Collection<AbstractImporter> importers)
         }
     }
 
+    @Override
+    public boolean canAcceptAssignments()
+    {
+        return !m_stopping;
```
Comment (akhanzode) [inline src/frontend/org/voltdb/importer/ImporterLifeCycleManager.java]: will there be a race between set and get here being on different threads?

**Comment by akhanzode** [comment]
Can we add a junit for channels getting redelivered?

File: src/frontend/org/voltdb/importer/ChannelChangeCallback.java (hunk 1 above)
Comment (JadenW) [inline src/frontend/org/voltdb/importer/ChannelChangeCallback.java]: Java 8 start support define default method for interface. https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html

This method is only override of ImporterLifeCycleManger

Comment (akhanzode) [inline src/frontend/org/voltdb/importer/ChannelChangeCallback.java]: oh ok nice.

**Comment by vtkstef** [comment]
--squash merged

**Pull request closed without merging**